pkgs/top-level/all-packages.nix

```diff
@@ -16678,7 +16678,8 @@ with pkgs;
 
   ograc = callPackage ../development/tools/rust/ograc { };
 
-  opensmalltalk-vm = callPackage ../development/compilers/opensmalltalk-vm { };
+  inherit (callPackages ../development/compilers/opensmalltalk-vm { })
+    squeak-cog-spur squeak-stack-spur newspeak-cog-spur;
```

Comment:
My personal opinion is that the `opensmalltalk-vm` namespace is nice to keep for discoverability, which can be achieved by using `recurseIntoAttrs`.

Reply:
Flattening makes the VMs discoverable by `callPackage`. If a package has the argument `opensmalltalk-vm` then the VM that the package uses can't be adjusted later with `override`. Unless  packages that depend on a VM in the attrset have their VM selected for them at `all-packages.nix` with a `callPackge … { inherit (opensmalltalk-vm) …; }` expression.

Reply:
@SuperSandro2000 do we have a policy on attrsets of packages in all-packages.nix?

Reply:
hmmm, not sure actually